letsencrypt/client.py

```diff
@@ -105,7 +105,8 @@ def register(config, account_storage, tos_cb=None):
                    "--register-unsafely-without-email was not present.")
             logger.warn(msg)
             raise errors.Error(msg)
-        logger.warn("Registering without email!")
+        if not config.dry_run:
```

Comment:
Channeling Seth a little here, there are subtleties to this:
* Should we also suppress this message in staging mode?  Or do we expect people to know about their staging accounts?
* If not, should we note when what we're registering for is a staging account?